src/core/IO/PortaudioDriver.cpp

```diff
@@ -76,6 +80,26 @@ int PortAudioDriver::init( unsigned nBufferSize )
 	return 0;
 }
 
+// List devices
+QStringList PortAudioDriver::getDevices() {
+	if ( ! m_bInitialised ) {
```

Comment:
This seems to cause problems on Linux if PortAudio wants to wait on the underlying ALSA device but it's in use by one of the other drivers. Yuck.

Reply:
It's starting to look like to safely query one driver about its available devices, we'll have to actively stop the current audio system to start the other one. While this is a bit cleaner in that it allows the listing method to be non-static (and virtual for some refactoring of the preferences dialog), it means that selecting a new audio system and device means *two* audio restarts.